src/parameters.rs

```diff
@@ -5,17 +5,61 @@
+//!
+//! Values are sourced from throughout the paper \[2\], especially Figure 1 (page 5) and Table 2
+//! (page 69).
+//!
+//! ## References
+//! 1. Recommendation for Key Management, Special Publication 800-57 Part 1 Rev. 5, NIST, 05/2020.
+//!    Interpreted via [keylength.org](https://www.keylength.com/en/compare/).
+//! 2. UC Non-Interactive, Proactive, Threshold ECDSA with Identifiable Aborts. Ran Canetti,
+//!    Rosario Gennaro, Steven Goldfeder, Nikolao Makriyannis, Udi Peled. EPrint version, Oct 2021.
+//!    [PDF](https://eprint.iacr.org/2021/060.pdf)
 
-/// See caption for Table 2 at bottom of page 69 in https://eprint.iacr.org/2021/060.pdf
-/// ELL = 1 * SecParam
-/// ELL_PRIME = 5 * SecParam
-/// EPSILON = 2 * SecParam
-pub(crate) const ELL: usize = 256;
-pub(crate) const ELL_PRIME: usize = 5 * 256;
-pub(crate) const EPSILON: usize = 2 * 256;
-pub(crate) const PRIME_BITS: usize = 512; // 512-bit primes are needed
-pub(crate) const SOUNDNESS_PARAMETER: usize = 8; // Needs to be a multiple of 8 for Pi_prm
+/// Security parameter κ defines the message size of an elliptic curve group element, and is also
+/// used to derive other parameter sizes.
+const SECURITY_PARAM: usize = 256;
+
+/// One of the ranges in proofs with a range check.
+///
+/// Specifically, it defines the range for the plaintext value of a Paillier ciphertexts (in
+/// [Π-enc](crate::zkp::pienc), [Π-log*](crate::zkp::pilog)) or of a group commitment
+/// (in [Π-aff-g](crate::zkp::piaffg)).
+/// The range value is `+/- 2^ELL`.
+pub(crate) const ELL: usize = SECURITY_PARAM;
+
+/// Defines one of the ranges in proofs with a range check.
+///
+/// Specifically, it defines the range for the plaintext value of a Paillier ciphertext in
+/// [Π-aff-g](crate::zkp::piaffg).
+/// The range value is `+/- 2^ELL_PRIME`.
+pub(crate) const ELL_PRIME: usize = 5 * SECURITY_PARAM;
+
+/// The flex space of a range check.
+///
+/// When a prover has a secret input x in the range `+/- 2^l`, the range check proofs
+/// guarantee to the verifier that `x` is in the range `+/- 2^(l + EPSILON)`.
+/// The same `EPSILON` value is used for all ranges -- those defined by both [`ELL`] and
+/// [`ELL_PRIME`].
+///
+/// It is part of the completeness bound in the range check proofs
+/// ([Π-enc](crate::zkp::pienc), [Π-log*](crate::zkp::pilog), and [Π-aff-g](crate::zkp::piaffg)).
+pub(crate) const EPSILON: usize = 2 * SECURITY_PARAM;
+
+/// Bit size of the (safe) prime factors of a Paillier modulus.
+///
+/// The product of two 1024-bit primes will produce 2048-bit Pallier moduli.
+pub(crate) const PRIME_BITS: usize = 1024;
+
+/// Number of repetitions required for statistical security in proofs that allow an adversary to
+/// guess a challenge value correctly with probability 1/2.
+///
+/// Historical documentation note without a verified source:
+/// > Needs to be a multiple of 8 for Pi_prm
+pub(crate) const SOUNDNESS_PARAMETER: usize = 8;
```

Comment:
I haven't yet figured out where the multiple of 8 part comes from, but it looks like SOUNDNESS_PARAMETER should be 80 instead of 8? (can be addressed later)

Reply:
Yes, 8 must have been a temporary choice for convenience or something.

Reply:
Yeah, I agree. Made #73 to fix.